test/functional/GUIDELINES.md

```diff
@@ -0,0 +1,80 @@
+# Preface
+
+This document contains guidlines and BKM for writing tests. Following these
+will significantly reduce numbers of comments you will receive on the code
+review and will speed up process of merging your patch.
+
+# Tests structure
+
+- All the tests reside in `tests/` directory, which contains subdirectories
+corresponding to individual test groups.
+- Test gruup directories contain tests that belong to particular group.
+Each test should belong to exactly one test group.
+- Each test shoud be placed in separate file (although test may comprise many
+test cases).
+
+# Test file
+
+- A test file corresponds to individual test.
+- Test file name should be preceded with `test_` prefix, which allows pytest
+to collect it using standard test discovery method.
+- Test file should contain one or more test cases.
+- Each test case is an function with the name preceded with `test_` prefix.
+- Test file can also contain another functions, classes etc. used by the test
+cases.
+
+# Test case
+
+A test case is a single function. There are four fundamental elements of each
+test function:
+1. Function name
+2. Docstring
+3. Pytest marks
+4. Testing code
+
+Following sections describe guidlines for each of these elements.
+
+## Function name
+
+- The name of the test case function should begin with `test_` prefix.
+- The name should be consise and meaningful in terms of what the test case
+actually does.
+
+## Docstring
+
+Each test case should contain yaml docstring in the following format:
+```python
+def test_example():
+    """
+    title: Brief test title
+    description: |
+      More detailed test description.
+      This is second line of test description.
+    pass_criteria:
+      - First pass criterium
+      - Second pass criterium
+      - Third pass criterium
+    """
+```
+
+## Pytest marks
+
+Test cases can use both standard pytest marks like `@pytest.mark.parametrize`
+as well as marks introduced by test framework like `@pytest.mark.require_disk`.
+
+Please refer to Test Framework documentation (coming soon).
+
+## Testing code
+
+- Typical test case comprises three basic elements: preparation, testing
+logic, and cleanup.
+- Test Framework allows to define test steps, step groups and iterations.
+- Every single line of test code must belong to some test step.
+- Test steps may be embedded into test step grups and iteration.
+- Test grups may be embedded into another test groups and iterations.
```

Comment:
other test groups
(another is singular)

Reply:
Fixed.